### ISSUE TITLE
AWX Inventory sync from VMware broken since 3.0.0.0

### ISSUE THREAD

**Issue by calvinbui** [ansible/awx#3088]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

 - API
 - UI

##### SUMMARY

Inventory sync does not work anymore from VMware vCenter/ESXi

##### ENVIRONMENT
* AWX version: 3.0.0 and devel
* AWX install method: docker on linux
* Ansible version:  2.7.6
* Operating System: Ubuntu 18.04
* Web Browser: Chrome

##### STEPS TO REPRODUCE

Create an inventory with source VMware vCenter and attempt to sync with the host.

##### EXPECTED RESULTS

Inventory synced without any issues

##### ACTUAL RESULTS

```
    1.463 INFO     Updating inventory 2: VMware Ubuntu Hosts
    1.480 INFO     Reading Ansible inventory source: /var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/plugins/inventory/vmware_inventory.py
    1.481 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    1.481 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    1.482 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python2.7/site-packages:
    4.698 INFO     Processing JSON output...
    4.699 INFO     Loaded 3 groups, 7 hosts
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/bin/awx-manage", line 11, in <module>
    load_entry_point('awx==3.0.0.0', 'console_scripts', 'awx-manage')()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/__init__.py", line 150, in manage
    execute_from_command_line(sys.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1030, in handle
    raise exc
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 975, in handle
    self.load_into_database()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 824, in load_into_database
    self._delete_hosts()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 413, in _delete_hosts
    instance_ids = all_instance_ids[offset:(offset + self._batch_size)]
TypeError: 'dict_keys' object is not subscriptable
```

##### ADDITIONAL INFORMATION

While the source is VMware vCenter, I have been syncing with VMware ESXi without any issues.

I am using the following filter, but it still fails with it removed:

```
---

host_filters: "{{ config.guestid == 'ubuntu64Guest' }}"
```

**Comment by wenottingham** [comment]
fixed here: https://github.com/ansible/awx/pull/3058

**Comment by calvinbui** [comment]
Running from the `devel` branch didn't solve the problem

**Comment by ryanpetrello** [comment]
@calvinbui on `devel`, are you still encountering the same traceback as above ending in:

```
    output = self.handle(*args, **options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1030, in handle
    raise exc
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 975, in handle
    self.load_into_database()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 824, in load_into_database
    self._delete_hosts()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 413, in _delete_hosts
    instance_ids = all_instance_ids[offset:(offset + self._batch_size)]
TypeError: 'dict_keys' object is not subscriptable
```

Can you provide the latest error/traceback you're encountering?

**Comment by calvinbui** [comment]
Yep, as i mentioned in the initial bug report and comment.

```
    1.386 INFO     Updating inventory 2: VMware Ubuntu Hosts
    1.402 INFO     Reading Ansible inventory source: /var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/plugins/inventory/vmware_inventory.py
    1.403 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    1.403 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    1.403 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python2.7/site-packages:
    5.547 INFO     Processing JSON output...
    5.548 INFO     Loaded 3 groups, 7 hosts
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/bin/awx-manage", line 11, in <module>
    load_entry_point('awx==3.0.0.0', 'console_scripts', 'awx-manage')()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/__init__.py", line 150, in manage
    execute_from_command_line(sys.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 1030, in handle
    raise exc
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 975, in handle
    self.load_into_database()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 824, in load_into_database
    self._delete_hosts()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 413, in _delete_hosts
    instance_ids = all_instance_ids[offset:(offset + self._batch_size)]
TypeError: 'dict_keys' object is not subscriptable
```

**Comment by ryanpetrello** [comment]
@calvinbui 

>   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/management/commands/inventory_import.py", line 413, in _delete_hosts

You're not running `devel` (at least not in the traceback you just pasted - here's `L413` of that file in `HEAD` of `devel`):

https://github.com/ansible/awx/blob/0e5e23372d80da0503dbe2852f66566b077f000e/awx/main/management/commands/inventory_import.py#L413

You're still running `3.0.0` code:

https://github.com/ansible/awx/blob/99288a5e184f01ec8c50f2ffee1394718f14c92f/awx/main/management/commands/inventory_import.py#L413

**Comment by calvinbui** [comment]
Thanks @ryanpetrello

I've re-cloned the repo, deleted all the docker containers + images and ran the playbook again but still getting the same problem.

```
git clone https://github.com/ansible/awx.git
docker rm -f $(docker ps -q -a)
docker image rm $(docker image ls -q -a)
cd awx/installer
ansible-playbook -i inventory install.yml -e "postgres_data_dir=/var/lib/postgresql/data"
```

Am I missing something to get `devel` running? Is the python3.6 venv using `3.0.0` somehow 🤔

**Comment by ryanpetrello** [comment]
@calvinbui we haven't released a new awx version beyond 3.0.0 that fixes this issue, so if you use the official images, you're going to continue to have problems.  If you want to build from `devel` instead of official images, please see:

https://github.com/ansible/awx/blob/devel/INSTALL.md#official-vs-building-images

**Comment by calvinbui** [comment]
Thanks for the link.

*Building* the images on `devel` now works

```

    1.506 INFO     Updating inventory 2: VMware Ubuntu Hosts
    1.516 INFO     Reading Ansible inventory source: /var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/plugins/inventory/vmware_inventory.py
    1.517 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    1.518 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    1.518 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python2.7/site-packages:
    5.641 INFO     Processing JSON output...
    5.642 INFO     Loaded 3 groups, 7 hosts
2019-01-28 20:45:26,663 DEBUG    awx.main.models.inventory Going to update inventory computed fields, pk=2
2019-01-28 20:45:26,697 DEBUG    awx.main.models.inventory Finished updating inventory computed fields, pk=2, in 0.032 seconds
    5.891 INFO     Inventory import completed for ESXi Ubuntu Guest Filter in 4.4s
```
